feat: Onboarding redesign

josheleonard opened the pull request:
<!-- Add brave-browser issue bellow that this PR will resolve -->
Resolves https://github.com/brave/brave-browser/issues/22591
Fixes: https://github.com/brave/brave-browser/issues/22817

Security review request: https://github.com/brave/security/issues/951

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Prep:
1. Reset Wallet via wallet settings
2. Open wallet via top-bar icon
3. Wallet Onboarding Welcome Panel should appear
4. Click "Learn more"
5. The Wallet Onboarding Welcome Tab should open

https://user-images.githubusercontent.com/30185185/177829416-ed0e5c30-8e3b-439a-a776-8471be90b8f7.mov


Scenario 1 - Learn More about wallets
1. Click "Learn more about crypto wallet"
2. A new tab should open to "https://brave.com/learn/what-is-crypto-wallet/"

https://user-images.githubusercontent.com/30185185/177829569-04a4c7b4-6310-4432-9431-d5762ceff1c0.mov



Scenario 1a - New Wallet
1. Click "Create a new wallet"
2. Agree to Terms
3. Create a password
4. Click "Got it"
5. Backup recovery phrase (check that reveal, clipboard and file download works)
7. Verify recovery phrase
8. Click Next
9. Onboarding success screen appears
10. Click Done
11. Wallet Portfolio page is opened

https://user-images.githubusercontent.com/30185185/177830113-3495d21d-5626-4548-9cc7-717d4daf218c.mov


Scenario 1b - Restore Wallet from recovery phrase

https://user-images.githubusercontent.com/30185185/177831188-eaf04550-b805-468a-ac6e-ba70f4a18a4b.mov

Scenario 1c - Import Wallet from MetaMask

https://user-images.githubusercontent.com/30185185/177862454-448104bc-2441-418e-991a-5a302bc596ba.mov


Scenario 1d - Import Wallet from Legacy Crypto Wallets

https://user-images.githubusercontent.com/30185185/177862819-35b7e355-25ee-48d3-8e08-f24b3a446619.mov

Scenario 2a - Buy Crypto

https://user-images.githubusercontent.com/30185185/177863360-09044e7d-4d01-4063-a9d1-4b866938f19c.mov

Scenario 2b - Fund Wallet with Crypto

https://user-images.githubusercontent.com/30185185/177863699-f84b140f-eee0-4f21-9528-b2b1ab31449a.mov

supermassive commented on components/brave_wallet_ui/common/async/__mocks__/lib.ts:
```diff
@@ -74,3 +74,12 @@ export const findUnstoppableDomainAddress = async (address: string) => {
+export const isStrongPassword = (value: string) => {
+  return (
+    (value.length >= 7) && // is at least 7 characters
+    /[-’/`~!#*$@_%+=.,^&(){}[\]|;:”<>?\\]/g.test(value) && // contains a special character
```
Most likely `’”` should be replaced with `'"`

josheleonard replied:
updated

thypon commented on components/brave_wallet_ui/page/screens/onboarding/backup-recovery-phrase/onboarding-backup-recovery-phrase.tsx:
```diff
@@ -0,0 +1,172 @@
+              {getLocale('braveWalletRecoveryPhraseBackupWarning')}
+              <LinkText
+                href='https://brave.com/learn/wallet-recovery-phrase/'
+                target='_blank'
```
Should we add `rel="noreferrer"` to every `target='_blank'` link?

thypon commented on components/brave_wallet_ui/page/screens/onboarding/welcome/onboarding-welcome.tsx:
```diff
@@ -0,0 +1,55 @@
+          url={WalletRoutes.Restore}
+        />
+
+        <LearnMoreLink href='https://brave.com/learn/what-is-crypto-wallet/' target='_blank'>{getLocale('braveWalletWhatIsACryptoWallet')}</LearnMoreLink>
```
Should we add `rel="noreferrer"` to every `target='_blank'` link?

thypon commented on components/brave_wallet_ui/page/screens/onboarding/onboarding-success/components/article-link-bubble/article-link-bubble.tsx:
```diff
@@ -0,0 +1,47 @@
+  return (
+    <Bubble
+      href={url}
+      target='_blank'
```
same here `rel=noreferrer`

josheleonard replied:
added

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/5d86ab84bb319e63cf6dc4e33558dd6e6030bdc3/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/b0e54762fe8e500a33757d3ad662612a78ed51f4/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/891f26c96fc01904d6bdbcf566be2790ffd104d1/storybook/index.html)

Douglashdaniel commented on components/brave_wallet_ui/common/hooks/use-multi-chain-buy-assets.ts:
```diff
@@ -0,0 +1,131 @@
+  const { getAllBuyAssets, getBuyAssetUrl } = useLib()
+
+  // state
+  const [buyAmount, setBuyAmount] = React.useState('')
```
`React.useState<string>('')`

josheleonard replied:
added explicit type definition

Douglashdaniel commented on components/brave_wallet_ui/common/hooks/use-multi-chain-buy-assets.ts:
```diff
@@ -0,0 +1,131 @@
+  })
+
+  // memos
+  const buyAssetNetworks = React.useMemo(() => {
```
Needs return type defined.

Douglashdaniel commented on components/brave_wallet_ui/common/hooks/use-multi-chain-buy-assets.ts:
```diff
@@ -0,0 +1,131 @@
+    )
+  }, [networkList])
+
+  const selectedAssetNetwork = React.useMemo(() => {
```
Needs return type defined.

josheleonard replied:
added return type to memos

Douglashdaniel commented on components/brave_wallet_ui/common/hooks/use-multi-chain-buy-assets.ts:
```diff
@@ -0,0 +1,131 @@
+    return selectedAsset ? getNetworkInfo(selectedAsset.chainId, selectedAsset.coin, buyAssetNetworks) : undefined
+  }, [selectedAsset, buyAssetNetworks])
+
+  const selectedAssetBuyOptions = React.useMemo(() => {
```
Needs return type defined.

josheleonard replied:
added return type to memos

josheleonard commented on components/brave_wallet/browser/brave_wallet_constants.h:
```diff
@@ -165,6 +181,8 @@ constexpr webui::LocalizedString kLocalizedStrings[] = {
     {"braveWalletToolTipCopiedToClipboard",
      IDS_BRAVE_WALLET_TOOL_TIP_COPIED_TO_CLIPBOARD},
+    {"braveWalletPasteFromClipboard",
+     IDS_BRAVE_WALLET_TOOL_TIP_PASTE_FROM_CLIPBOARD},
```
Thanks. Fixed

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/106f7a313e342e8c0652929e61f0f6b770a70227/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/4458d29e6269ea6815a29be204d2a3c4ebdd0f12/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/5db40242f48fb60061f4aabdc8198ba89375e5b3/storybook/index.html)

muliswilliam commented:
Found a bug when selecting recovery phrase words:

https://user-images.githubusercontent.com/48117473/177186761-f7412282-647e-4f4c-a2d3-6a6757d1fc54.mov

muliswilliam commented:
Recovery Phrase font color is clear
<img width="571" alt="Screenshot 2022-07-04 at 18 50 05" src="https://user-images.githubusercontent.com/48117473/177187915-d749968c-d044-49d3-8f93-4746e441ed90.png">

Douglashdaniel commented on components/brave_wallet_ui/components/buy-send-swap/buy/index.tsx:
```diff
@@ -80,9 +95,14 @@ function Buy (props: Props) {
-    setBuyOptions(supportingBuyOptions)
   }, [selectedAsset, wyreAssetOptions, rampAssetOptions])
 
+  const isSelectedNetworkSupported = React.useMemo(() => {
```
`:boolean` return type

josheleonard replied:
added explicit return type definition

Douglashdaniel commented on components/brave_wallet_ui/components/buy-send-swap/create-account/index.tsx:
```diff
@@ -28,51 +30,108 @@ import {
   const dispatch = useDispatch()
 
+  // state
+  const [showUnlock, setShowUnlock] = React.useState(false)
```
`React.useState<boolean>(false)`

josheleonard replied:
added explicit type definition

Douglashdaniel commented on components/brave_wallet_ui/components/buy-send-swap/create-account/index.tsx:
```diff
@@ -28,51 +30,108 @@ import {
+  const [showUnlock, setShowUnlock] = React.useState(false)
+
+  // memos
+  const accountNetwork = React.useMemo(() => {
```
`BraveWallet.NetworkInfo` return type

josheleonard replied:
added explicit type definition

Douglashdaniel commented on components/brave_wallet_ui/components/desktop/views/accounts/account.tsx:
```diff
@@ -94,7 +95,10 @@ export const Account = (props: Props) => {
+  const {
+    isCopied: copied,
+    temporaryCopyToClipboard: copyText
+  } = useTemporaryCopyToClipboard(1500)
```
Is `1 1/2` seconds long enough for someone to copy and paste their wallet address somewhere?
Curious why we need this for addresses?

josheleonard replied:
1.5 seconds was the time set by @muliswilliam 's other clipboard hook. 

@muliswilliam, you ok with me increasing the time for this?

Douglashdaniel replied:
The difference here though is @muliswilliam `1.5` seconds was for displaying `Copied` tooltip.

This now uses that time to clear the `clipboard` https://github.com/brave/brave-core/pull/13146/files#diff-1524a199b29529703bbec790905dc4ce35f2f8842dd5aa033f4b08baac1a95aaR34

Maybe it should be separated or have an optional timeout prop to clear the clipboard if provided.

josheleonard replied:
Fixed in: [61327a9](https://github.com/brave/brave-core/pull/13146/commits/61327a9c2e6b3c9f3483087155c9263a431817f3)

Douglashdaniel commented on components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-overview.tsx:
```diff
@@ -231,14 +227,24 @@ export const PortfolioOverview = () => {
-        onSetFilteredAssetList={setfilteredAssetList}
-        onSelectAsset={onSelectAsset}
-        hideBalances={hideBalances}
+        renderToken={(item) =>
```
Why is this being rendered through a prop?

josheleonard commented:
> Found a bug when selecting recovery phrase words:
> 
>  Recovery.phrase.bug.mov

This is not a bug. The _first_, _third_, and _last_ words need to be selected. (you selected the first 3 words)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/b98fb8be7d4d34282805f21d98d29a21d4eb8083/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/400b02b111084eb995eb3167e15bd3934ceab864/storybook/index.html)

josheleonard commented:
> Recovery Phrase font color is clear <img alt="Screenshot 2022-07-04 at 18 50 05" width="571" src="https://user-images.githubusercontent.com/48117473/177187915-d749968c-d044-49d3-8f93-4746e441ed90.png">

Fixed as of [314587c](https://github.com/brave/brave-core/pull/13146/commits/314587c794dd73d09ac5fb0cbba126823a599fc6)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/fe70387afc3f146dc91e25be1d20bcf594974c00/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/1e0853097c622f6d032d90c76e63858f90d6564a/storybook/index.html)

Douglashdaniel commented on components/brave_wallet_ui/components/extension/lock-panel/style.ts:
```diff
@@ -3,14 +3,14 @@ import SecureIcon from '../../../assets/svg-icons/onboarding/secure-your-crypto.
   align-items: center;
   justify-content: center;
-  background-color: ${(p) => p.theme.color.background01};
+  background-color: ${(p) => p.hideBackground ? 'transparent' : p.theme.color.background01};
```
Curious why we need to hide the lock panels `background`?

josheleonard replied:
So it can be rendered inside of another screen. I am now rendering this inside of create-account component since the wallet needs to be unlocked for account creation (edge case when buying tokens during onboarding)

Douglashdaniel commented on components/brave_wallet_ui/components/desktop/centered-page-layout/centered-page-layout.style.ts:
```diff
@@ -0,0 +1,29 @@
+
+import styled from 'styled-components'
+
+export const StyledWrapper = styled('div') <{}>`
```
Do we need this empty `prop` object?

josheleonard replied:
removed

Douglashdaniel commented on components/brave_wallet_ui/components/desktop/centered-page-layout/centered-page-layout.style.ts:
```diff
@@ -0,0 +1,29 @@
+  overflow-y: scroll;
+ `
+
+export const StyledContent = styled('div') <{}>`
```
ditto

josheleonard replied:
removed

Douglashdaniel commented on components/brave_wallet_ui/components/shared/buy-option/buy-asset-option.tsx:
```diff
@@ -0,0 +1,93 @@
+        .replace('$2', tokenNetwork.chainName ?? '')
+    }
+    return token.symbol
+  }, [tokenNetwork, token])
```
Missing `isPanel` dependency.

Douglashdaniel commented on components/brave_wallet_ui/components/shared/buy-option/buy-asset-option.tsx:
```diff
@@ -0,0 +1,93 @@
+  isPanel
+}: Props) => {
+  // memos
+  const networkDescription = React.useMemo(() => {
```
returns a `string`

Douglashdaniel commented on components/brave_wallet_ui/components/shared/checkbox/checkbox.style.tsx:
```diff
@@ -0,0 +1,103 @@
+    --checkbox-box-size: ${p.size === 'big' ? 24 : 18}px;
+    --checkbox-label-size: ${p.size === 'big' ? 16 : 14}px;
+    --checkbox-box-spacing: ${p.size === 'big' ? 17 : 12}px;
+    --checkbox-box-color: #696FDC;
+    --checkbox-label-color: #B8B9C4;
```
Are these available in `theme`?

Same for the bellow in this file.

josheleonard replied:
I was unable to find them using the `theme` intellisense

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/efbaffc6d1d9780689993189910f276264fa8761/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/d232de082a8c156a56345bc18b6bceb18c827174/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/fb1463b2bbd47ffc0d3426c597891b8070415e2b/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/405bfeb0c31599c2854422ca8191be490664c605/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/7ed18e729f572a4e3d8ca2970a345682a6f8b757/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/b111aec67d1311211078a3d6ab292e7ac2fc1d04/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/c1b678dd9a411a1eec9b1fd9d2a9962a852a901f/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/8a306239f6ba531f133e8c4a9d5564974e006387/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/6c48cd884d0fdc8162e5d60d36b590c257c97ab9/storybook/index.html)

Douglashdaniel reviewed (commented):
Getting this console warning creating a password.
<img width="1552" alt="Screen Shot 2022-07-14 at 3 04 48 PM" src="https://user-images.githubusercontent.com/40611140/179086038-93134831-241f-48a4-927b-d5fd20719488.png">

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/7bf3d9df5b31b5398ba0166e8564b293344b8b8c/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/d80c27b68e7ec0a212514dc24e0aac1b0ffff8bf/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/ee7961d17998ec1e1c0a0fa41ba3cad4c54fc783/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/dd97a688b887c4e5b981ac2e029c28e8f81d0b61/storybook/index.html)

josheleonard commented on components/brave_wallet/browser/brave_wallet_constants.cc:
```diff
@@ -14,9 +14,6 @@ const std::vector<mojom::BlockchainToken>& GetWyreBuyTokens() {
        {"0x0D8775F648430679A709E98d2b0Cb6250d2887EF", "Basic Attention Token",
         "bat.png", true, false, "BAT", 18, true, "", "", mojom::kMainnetChainId,
         mojom::CoinType::ETH},
-       {"0x4829043Fc625FdA139523b2ceB99D95354e2b359", "Avalanche C-Chain",
```
0x4829043Fc625FdA139523b2ceB99D95354e2b359 was not actually the C-chain token

josheleonard commented on components/brave_wallet/browser/brave_wallet_constants.cc:
```diff
@@ -101,8 +98,7 @@ const std::vector<mojom::BlockchainToken>& GetRampBuyTokens() {
         mojom::kAvalancheMainnetChainId, mojom::CoinType::ETH},
-       {"0x82030cdbd9e4b7c5bb0b811a61da6360d69449cc", "RealFevr", "", true,
-        false, "FEVR", 18, true, "", "",
+       {"", "Binance Smart Chain", "", false, false, "BNB", 18, true, "", "",
```
BNB was missing from the buy-assets list

josheleonard commented on components/brave_wallet_ui/common/async/lib.ts:
```diff
@@ -185,6 +187,16 @@ export async function getBuyAssetUrl (args: {
+  // adjust Wyre on-ramp url for multichain
+  if (args.onRampProvider === BraveWallet.OnRampProvider.kWyre) {
+    if (args.chainId === BraveWallet.AVALANCHE_MAINNET_CHAIN_ID) {
+      return url.replace('dest=ethereum:', 'dest=avalanche:')
```
Allowing buying assets directly on Avalanche from Wyre

josheleonard commented on components/brave_wallet_ui/common/async/lib.ts:
```diff
@@ -185,6 +187,16 @@ export async function getBuyAssetUrl (args: {
+      return url.replace('dest=ethereum:', 'dest=avalanche:')
+    }
+    if (args.chainId === BraveWallet.POLYGON_MAINNET_CHAIN_ID) {
+      return url.replace('dest=ethereum:', 'dest=matic:')
```
Allowing buying assets directly on Polygon from Wyre

josheleonard commented on components/brave_wallet_ui/common/hooks/use-multi-chain-buy-assets.ts:
```diff
@@ -0,0 +1,136 @@
+      })
+  }, [getAllBuyAssets, isMounted])
+
+  const openBuyAssetLink = React.useCallback(({ buyOption, depositAddress }: {
```
opens link to on-ramp partner for asset purchase

josheleonard commented on components/brave_wallet_ui/common/hooks/use-password-strength.ts:
```diff
@@ -4,23 +4,37 @@
+  // methods
+  const checkIsStrongPassword = React.useCallback((pass: string) => {
+    // is at least 8 characters
+    const isLongEnough = pass.length >= 8
```
Wallet was using an obsolete password policy
https://github.com/brave/brave-browser/issues/22817

josheleonard commented on components/brave_wallet_ui/common/hooks/use-password-strength.ts:
```diff
@@ -4,23 +4,37 @@
+    // granular results of password strength check
+    return {
+      isLongEnough,
+      isStrongPassword: isLongEnough
```
This is now the only requirement

josheleonard commented on components/brave_wallet_ui/common/hooks/use-password-strength.ts:
```diff
@@ -4,23 +4,37 @@
+    return checkIsStrongPassword(password)
+  }, [checkIsStrongPassword, password])
+
+  const isStrongPassword = React.useMemo(() => {
+    return passwordStrength.isLongEnough
+  }, [passwordStrength.isLongEnough])
```
This is now the only requirement

josheleonard commented on components/brave_wallet_ui/components/buy-send-swap/create-account/index.tsx:
```diff
@@ -28,55 +30,115 @@ import {
+      dispatch(WalletActions.addAccount({
+        accountName: suggestedAccountName,
+        coin: accountNetwork.coin
+      }))
     }
+
     if (isPanel) {
       dispatch(PanelActions.navigateTo('main'))
     }
-  }, [selectedNetwork, suggestedAccountName, isPanel])
+  }, [accountNetwork, suggestedAccountName, isPanel, isWalletLocked, showUnlock])
+
+  const handleUnlockAttempt = React.useCallback((password: string): void => {
+    dispatch(WalletActions.unlockWallet({ password }))
+    dispatch(WalletActions.addAccount({
+      accountName: suggestedAccountName,
+      coin: accountNetwork.coin
+    }))
+  }, [suggestedAccountName, accountNetwork])
+
+  // effects
+  React.useEffect(() => {
+    // hide unlock screen on unlock success
+    if (!isWalletLocked && showUnlock) {
+      setShowUnlock(false)
+    }
+  }, [isWalletLocked, showUnlock])
+
+  // render
+  if (isWalletLocked && showUnlock) {
+    return <StyledWrapper>
+      <Description style={{ fontSize: 16 }}>
+        {getLocale('braveWalletUnlockNeededToCreateAccount')}
+      </Description>
+      <LockPanel
+        hideBackground
+        onSubmit={handleUnlockAttempt}
+      />
+    </StyledWrapper>
+  }
```
Wallet can now be unlocked from create-account component

josheleonard commented on components/brave_wallet_ui/components/extension/buttons/nav-button/index.tsx:
```diff
@@ -17,35 +23,78 @@ export type PanelButtonTypes =
+  }, [buttonType, text])
+
+  // render
+  return url ? (
+    <StyledLink
+      disabled={disabled}
+      buttonType={buttonType}
+      onClick={onSubmit}
+      to={url || ''}
+      maxHeight={maxHeight}
+      minWidth={minWidth}
+    >
+      {buttonContent}
+    </StyledLink>
+  ) : (
+    <StyledButton
```
The nav button can now be rendered as an "a" tag

josheleonard replied:
@thypon  @kdenhartog  check for HTML injection attacks

kdenhartog replied:
^ Not sure what I was thinking during the review to ask for this, but I don't see any concerns with HTML injection going on here. Looks fine to me

josheleonard commented on components/brave_wallet_ui/components/shared/password-input/new-password-input.tsx:
```diff
@@ -0,0 +1,188 @@
+    password,
+    setConfirmedPassword,
+    passwordsMatch
+  } = usePasswordStrength()
```
New password validation logic

josheleonard commented on components/brave_wallet_ui/components/shared/tooltip/password-strength-tooltip.tsx:
```diff
@@ -0,0 +1,76 @@
+      </PasswordStrengthHeading>
+
+      <CriteriaCheckContainer>
+        {isLongEnough && <GreenCheckmark />}
```
show checkmark when password is at least 8 characters

josheleonard commented on components/brave_wallet_ui/constants/types.ts:
```diff
@@ -575,10 +578,29 @@ export enum WalletRoutes {
 
   // onboarding
   Onboarding = '/crypto/onboarding',
+  OnboardingWelcome = '/crypto/onboarding/welcome',
+
+  // onboarding (new wallet)
   OnboardingCreatePassword = '/crypto/onboarding/create-password',
   OnboardingBackupWallet = '/crypto/onboarding/backup-wallet',
+  OnboardingExplainRecoveryPhrase = '/crypto/onboarding/explain-recovery-phrase',
+  OnboardingBackupRecoveryPhrase = '/crypto/onboarding/backup-recovery-phrase',
+  OnboardingVerifyRecoveryPhrase = '/crypto/onboarding/verify-recovery-phrase',
+
+  // onboarding (import / restore)
+  OnboardingImportOrRestore = '/crypto/onboarding/import-or-restore',
   OnboardingImportMetaMask = '/crypto/onboarding/import-metamask-wallet',
+  OnboardingImportMetaMaskSeed = '/crypto/onboarding/import-metamask-seed',
+  OnboardingRestoreWallet = '/crypto/onboarding/restore-wallet',
   OnboardingImportCryptoWallets = '/crypto/onboarding/import-legacy-wallet',
+  OnboardingImportCryptoWalletsSeed = '/crypto/onboarding/import-legacy-seed',
+
+  // onboarding complete
+  OnboardingComplete = '/crypto/onboarding/complete',
+
+  // fund wallet page
+  FundWalletPage = '/crypto/fund-wallet',
+  DepositFundsPage = '/crypto/deposit-funds',
```
New URLS

josheleonard commented on components/brave_wallet_ui/constants/types.ts:
```diff
@@ -604,6 +626,7 @@ export enum WalletRoutes {
 
   // portfolio
   Portfolio = '/crypto/portfolio',
+  PortfolioSub = '/crypto/portfolio/:id?',
```
New URL

josheleonard commented on components/brave_wallet_ui/constants/types.ts:
```diff
@@ -662,6 +685,15 @@ export const SupportedCoinTypes = [
   BraveWallet.CoinType.FIL
 ]
 
+export const SupportedOnRampNetworks = [
+  BraveWallet.SOLANA_MAINNET,
+  BraveWallet.MAINNET_CHAIN_ID, // ETH
+  BraveWallet.POLYGON_MAINNET_CHAIN_ID,
+  BraveWallet.BINANCE_SMART_CHAIN_MAINNET_CHAIN_ID,
+  BraveWallet.CELO_MAINNET_CHAIN_ID,
+  BraveWallet.AVALANCHE_MAINNET_CHAIN_ID
+]
```
Chains that our on-ramp providers can deposit assets to

josheleonard commented on components/brave_wallet_ui/page/async/wallet_page_async_handler.ts:
```diff
@@ -44,6 +46,41 @@ async function refreshWalletInfo (store: Store) {
   store.dispatch(WalletActions.initialized({ ...result, selectedAccount: '', visibleTokens: [] }))
 }
 
+async function importFromExternalWallet (
+  walletType: BraveWallet.ExternalWalletType,
+  payload: ImportFromExternalWalletPayloadType
+): Promise<ImportWalletErrorPayloadType> {
+  const {
+    braveWalletService,
+    keyringService
+  } = getWalletPageApiProxy()
+
+  const result = await braveWalletService.importFromExternalWallet(
+    walletType,
+    payload.password,
+    payload.newPassword
+  )
+
+  // complete backup if a new password was provided
+  if (payload.newPassword && result.success) {
+    keyringService.notifyWalletBackupComplete()
+  }
+
+  // was the provided import password correct?
+  const checkExistingPasswordError = result.errorMessage === getLocale('braveWalletImportPasswordError')
+    ? result.errorMessage
+    : undefined
+
+  // was import successful (if attempted)
+  const importError = payload.newPassword ? result.errorMessage || undefined : undefined
+
+  return {
+    hasError: !!(importError || checkExistingPasswordError),
+    errorMessage: importError || checkExistingPasswordError,
+    incrementAttempts: true
+  }
+}
```
Updated wallet import implementation so that we can check the import password before completing the import process

josheleonard commented on components/brave_wallet_ui/page/async/wallet_page_async_handler.ts:
```diff
@@ -60,7 +97,9 @@ handler.on(WalletPageActions.restoreWallet.getType(), async (store: Store, paylo
   await refreshWalletInfo(store)
   store.dispatch(WalletPageActions.setShowIsRestoring(false))
-  store.dispatch(WalletPageActions.walletSetupComplete())
+  if (payload?.completeWalletSetup) {
+    store.dispatch(WalletPageActions.walletSetupComplete(payload.completeWalletSetup))
+  }
```
We now have the option to create wallets without also exiting the onboarding flow

josheleonard commented on components/brave_wallet_ui/page/async/wallet_page_async_handler.ts:
```diff
@@ -164,6 +203,7 @@ handler.on(WalletPageActions.removeHardwareAccount.getType(), async (store: Stor
 })
 
 handler.on(WalletPageActions.checkWalletsToImport.getType(), async (store) => {
+  store.dispatch(WalletPageActions.setImportWalletsCheckComplete(false))
```
We can now show loading indicators while import-able wallets are being detected

josheleonard commented on components/brave_wallet_ui/page/async/wallet_page_async_handler.ts:
```diff
@@ -173,36 +213,23 @@ handler.on(WalletPageActions.checkWalletsToImport.getType(), async (store) => {
-  const braveWalletService = getWalletPageApiProxy().braveWalletService
-  const keyringService = getWalletPageApiProxy().keyringService
-  const result =
-    await braveWalletService.importFromExternalWallet(
-      BraveWallet.ExternalWalletType.MetaMask, payload.password, payload.newPassword)
-  if (result.success) {
-    keyringService.notifyWalletBackupComplete()
-  }
-  store.dispatch(WalletPageActions.setImportWalletError({
-    hasError: !result.success,
-    errorMessage: result.errorMessage ?? undefined
-  }))
+  const results: ImportWalletErrorPayloadType = await importFromExternalWallet(
+    BraveWallet.ExternalWalletType.MetaMask,
+    payload
+  )
+  store.dispatch(WalletPageActions.setImportWalletError(results))
```
Updated wallet import implementation so that we can check the import password before completing the import process

josheleonard commented on components/brave_wallet_ui/page/screens/onboarding/backup-recovery-phrase/onboarding-backup-recovery-phrase.tsx:
```diff
@@ -0,0 +1,170 @@
+import { OnboardingNewWalletStepsNavigation } from '../components/onboarding-steps-navigation/onboarding-steps-navigation'
+import { NavButton } from '../../../../components/extension'
+
+// storybook compiler thinks `randomUUID` doesnt exist
+const randomUUID = () => (
+  window.crypto as Crypto & { randomUUID: () => string }
+).randomUUID()
```
Using built-in crypto for randomness

josheleonard commented on components/brave_wallet_ui/page/screens/onboarding/backup-recovery-phrase/onboarding-backup-recovery-phrase.tsx:
```diff
@@ -0,0 +1,170 @@
+            </Description>
+          </TitleAndDescriptionContainer>
+
+          <PhraseCard>
+            <PhraseCardTopRow>
+              <ToggleVisibilityButton
+                isVisible={isPhraseShown}
+                onClick={toggleShowPhrase}
+              />
+            </PhraseCardTopRow>
+
+            <PhraseCardBody>
+              <RecoveryPhrase
+                hidden={!isPhraseShown}
+                onClickReveal={revealPhrase}
+                recoveryPhrase={recoveryPhrase}
+              />
+            </PhraseCardBody>
+
+            <PhraseCardBottomRow>
+
+              <CopyButton onClick={onCopyPhrase} />
+
+              <a
+                href={phraseDownloadUri}
+                download={`brave-wallet-private-key-${fileGUID}.txt`}
+                onClick={onDownloadPhraseFile}
+              >
+                <DownloadButton />
+              </a>
+
+              {isCopied &&
+                <>
+                  <CopiedToClipboardConfirmation />
+                  <HorizontalSpace space='52px' />
+                </>
+              }
+
+            </PhraseCardBottomRow>
+          </PhraseCard>
```
New recovery phrase backup implementation

josheleonard replied:
@kdenhartog check leaking thru file storage

check <a> tag

josheleonard commented on components/brave_wallet_ui/page/screens/onboarding/onboarding.routes.tsx:
```diff
@@ -0,0 +1,128 @@
+      <Route path={WalletRoutes.OnboardingImportMetaMask} exact>
+        <OnboardingRestoreFromRecoveryPhrase
+          key='metamask'
+          restoreFrom='metamask'
+        />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingImportMetaMaskSeed} exact>
+        <OnboardingRestoreFromRecoveryPhrase
+          key='metamask-seed'
+          restoreFrom='metamask-seed'
+        />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingImportCryptoWallets} exact>
+        <OnboardingRestoreFromRecoveryPhrase
+          key='legacy'
+          restoreFrom='legacy'
+        />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingImportCryptoWalletsSeed} exact>
+        <OnboardingRestoreFromRecoveryPhrase
+          key='legacy-seed'
+          restoreFrom='legacy-seed'
+        />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingWelcome} exact>
+        <OnboardingWelcome />
+      </Route>
+
+      <Route path={WalletRoutes.Onboarding} exact>
+        <Redirect to={WalletRoutes.OnboardingWelcome} />
+      </Route>
+
+      <Redirect to={WalletRoutes.Onboarding} />
+
+    </Switch>
+  )
```
Now requiring that wallet terms-of-use are agreed to before certain pages are accessible

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/02411f8c200ffb80e6c9cc81ed06635635cbd0d2/storybook/index.html)

josheleonard commented on components/brave_wallet_ui/page/screens/onboarding/restore-from-recovery-phrase/restore-from-recovery-phrase.tsx:
```diff
@@ -0,0 +1,508 @@
+          }
+
+          {currentStep === RestoreFromOtherWalletSteps.newPassword &&
+            <>
+              <NewPasswordInput
+                autoFocus={true}
+                onSubmit={restoreWallet}
+                onChange={handlePasswordChange}
+              />
+              {importWalletError?.hasError &&
+                <ErrorText>
+                  {importWalletError.errorMessage}
+                </ErrorText>
+              }
+            </>
+          }
+
+          {!isCheckingExtensions &&
+            <NextButtonRow>
+              <NavButton
+                buttonType='primary'
+                text={isCheckingImportPassword
+                  ? <LoadingIcon
+                    size='24px'
+                    opacity={0.8}
+                    color='interactive08'
+                  /> as unknown as string
+                  : getLocale('braveWalletButtonContinue')
+                }
+                onSubmit={onContinueClicked}
+                disabled={
+                  isCheckingImportPassword ||
+                  isImportingFromExtension && currentStep === RestoreFromOtherWalletSteps.currentPassword && !extensionPassword ||
+                  restoreFrom === 'seed' && !phraseInput ||
+                  currentStep === RestoreFromOtherWalletSteps.phrase && invalidMnemonic ||
+                  currentStep === RestoreFromOtherWalletSteps.newPassword && !isPasswordValid
+                }
+              />
+            </NextButtonRow>
+          }
```
New wallet restore implementation. Will now check the import password before allowing the creation of a new password

josheleonard replied:
@kdenhartog  recovery phrase used

josheleonard commented on components/brave_wallet_ui/page/screens/onboarding/welcome/onboarding-welcome.tsx:
```diff
@@ -0,0 +1,129 @@
+    if (!setupStillInProgress) {
+      dispatch(WalletPageActions.walletSetupComplete(false))
+    }
+    dispatch(WalletPageActions.onOnboardingShown())
```
existing p3a question - moved to this new screen

josheleonard commented on components/brave_wallet_ui/utils/password-utils.test.ts:
```diff
@@ -1,23 +0,0 @@
-import { isStrongPassword } from './password-utils'
-
-describe('Check if password is strong', () => {
-  test('Password meets all requirements except including a number.', () => {
-    const value = 'BraveRocks!'
-    expect(isStrongPassword.test(value)).toEqual(false)
-  })
-
-  test('Password meets all requirements except including a special character.', () => {
-    const value = 'BraveRocks1'
-    expect(isStrongPassword.test(value)).toEqual(false)
-  })
-
-  test('Password meets all requirements except being 7 characters long.', () => {
-    const value = 'Brav#1'
-    expect(isStrongPassword.test(value)).toEqual(false)
-  })
-
-  test('Password meets all requirements.', () => {
-    const value = 'Brave#1'
-    expect(isStrongPassword.test(value)).toEqual(true)
-  })
-})
```
This was testing a RexExp that is also now removed

josheleonard commented on components/brave_wallet_ui/utils/password-utils.ts:
```diff
@@ -1 +0,0 @@
-export const isStrongPassword = new RegExp('^(?=.*[0-9])(?=.*[\`\~\!\@\#$\%^\&*()+\=?\;\:|\<\>\,.\'\"{}\/[\_\\\]\\\\-])(?=.{7,})')
```
Now using a util function

josheleonard commented on components/brave_wallet_ui/utils/qr-code-utils.ts:
```diff
@@ -0,0 +1,18 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as qr from 'qr-image'
+
+export const generateQRCode = (data: string): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const image = qr.image(data)
+    let chunks: Uint8Array[] = []
+    image
+      .on('data', (chunk: Uint8Array) => chunks.push(chunk))
+      .on('end', () => {
+        resolve(`data:image/png;base64,${Buffer.concat(chunks).toString('base64')}`)
+      })
+  })
+}
```
Extracted QR code generation implementation to a helper function

josheleonard commented on components/brave_wallet_ui/utils/string-utils.ts:
```diff
@@ -13,22 +18,32 @@ export const isRemoteImageURL = (url?: string) =>
+    case BraveWallet.MAINNET_CHAIN_ID: return ''
+    default: return ''
+  }
+}
+
 export const getRampNetworkPrefix = (chainId: string) => {
   switch (chainId) {
-    case BraveWallet.MAINNET_CHAIN_ID:
-    case BraveWallet.AVALANCHE_MAINNET_CHAIN_ID:
-    case BraveWallet.CELO_MAINNET_CHAIN_ID:
-      return ''
-    case BraveWallet.BINANCE_SMART_CHAIN_MAINNET_CHAIN_ID:
-      return 'BSC'
-    case BraveWallet.POLYGON_MAINNET_CHAIN_ID:
-      return 'MATIC'
-    case BraveWallet.SOLANA_MAINNET:
-      return 'SOLANA'
-    case BraveWallet.OPTIMISM_MAINNET_CHAIN_ID:
-      return 'OPTIMISM'
-    default:
-      return ''
+    case BraveWallet.MAINNET_CHAIN_ID: return ''
+    case BraveWallet.AVALANCHE_MAINNET_CHAIN_ID: return 'AVAXC'
+    case BraveWallet.BINANCE_SMART_CHAIN_MAINNET_CHAIN_ID: return 'BSC'
+    case BraveWallet.POLYGON_MAINNET_CHAIN_ID: return 'MATIC'
+    case BraveWallet.SOLANA_MAINNET: return 'SOLANA'
+    case BraveWallet.OPTIMISM_MAINNET_CHAIN_ID: return 'OPTIMISM'
+    case BraveWallet.CELO_MAINNET_CHAIN_ID: return ''
+    default: return ''
```
Multi-chain purchases

josheleonard commented on components/brave_wallet_ui/components/desktop/network-filter-selector/index.tsx:
```diff
@@ -29,18 +29,28 @@ import {
+}
+
+export const NetworkFilterSelector = ({ networkListSubset }: Props) => {
+  // state
   const [showNetworkFilter, setShowNetworkFilter] = React.useState<boolean>(false)
 
   // redux
   const {
     selectedNetworkFilter,
-    networkList,
+    networkList: reduxNetworkList,
     isTestNetworksEnabled
   } = useSelector(({ wallet }: { wallet: WalletState }) => wallet)
 
   const dispatch = useDispatch()
 
+  // memos
+  const networkList = React.useMemo(() => {
+    return networkListSubset || reduxNetworkList
+  }, [networkListSubset, reduxNetworkList])
```
Now allowing custom networks list for filter

josheleonard commented on components/brave_wallet_ui/components/desktop/popup-modals/account-settings-modal/index.tsx:
```diff
@@ -89,15 +109,13 @@ const AddAccountModal = (props: Props) => {
-        setQRCode(`data:image/png;base64,${Buffer.concat(chunks).toString('base64')}`)
-      })
-  }
+  const generateQRData = React.useCallback(() => {
+    generateQRCode(account.address).then(qr => {
+      if (isMounted) {
+        setQRCode(qr)
+      }
+    })
+  }, [account, isMounted])
```
Using reusable helper function for this now

josheleonard commented on components/brave_wallet_ui/components/extension/lock-panel/index.tsx:
```diff
@@ -1,56 +1,99 @@
+  onSubmit,
+  onClickRestore,
+  disabled,
+  hideBackground
+}: Props) {
+  // redux
+  const dispatch = useDispatch()
+  const hasIncorrectPassword = useSelector(({
+    wallet
+  }: { wallet: WalletState }) => wallet.hasIncorrectPassword)
+
+  // state
+  const [inputValue, setInputValue] = React.useState<string>('')
 
-  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+  // methods
+  const handlePasswordChanged = React.useCallback((value: string) => {
+    setInputValue(value)
+    if (hasIncorrectPassword) {
+      dispatch(WalletActions.hasIncorrectPassword(false))
+    }
+  }, [hasIncorrectPassword])
+
+  const handleKeyDown = React.useCallback((event: React.KeyboardEvent<HTMLInputElement>) => {
     if (event.key === 'Enter' && !disabled) {
-      onSubmit()
+      onSubmit(inputValue)
     }
-  }
+  }, [disabled, onSubmit, inputValue])
+
+  const onClickUnlockButton = React.useCallback(() => {
+    onSubmit(inputValue)
+    setInputValue('')
+  }, [onSubmit, inputValue])
 
+  // render
   return (
-    <StyledWrapper>
+    <StyledWrapper hideBackground={hideBackground}>
```
Lock panel can now be rendered within other components.
This component now internally tracks it's own state (password input changes)

josheleonard commented on components/brave_wallet_ui/page/container.tsx:
```diff
@@ -149,13 +145,15 @@ export const Container = () => {
-      !walletLocation.includes(WalletRoutes.Onboarding) &&
-      walletLocation !== WalletRoutes.Unlock &&
-      walletLocation !== WalletRoutes.Restore // can be accessed from unlock screen
+      walletLocation.includes(WalletRoutes.Accounts) ||
+      walletLocation.includes(WalletRoutes.Backup) ||
+      walletLocation.includes(WalletRoutes.DepositFundsPage) ||
+      walletLocation.includes(WalletRoutes.FundWalletPage) ||
+      walletLocation.includes(WalletRoutes.Portfolio)
```
New list of URLs that we allow redirecting to after unlock

josheleonard commented on components/brave_wallet_ui/page/container.tsx:
```diff
@@ -164,87 +162,84 @@ export const Container = () => {
+                        onSubmit={unlockWallet}
+                        disabled={inputValue === ''}
+                        onPasswordChanged={handlePasswordChanged}
+                        hasPasswordError={hasIncorrectPassword}
+                        onShowRestore={onToggleShowRestore}
+                      />
+                    </SimplePageWrapper>
+                  </Route>
+                }
+
+                {!isWalletLocked &&
+                  <Route path={WalletRoutes.Backup} exact={true}>
+                    <SimplePageWrapper>
+                      <BackupWallet
+                        isOnboarding={false}
+                        onCancel={onHideBackup}
+                      />
+                    </SimplePageWrapper>
+                  </Route>
+                }
+
+                {!isWalletLocked &&
+                  <Route path={WalletRoutes.CryptoPage}>
+                    <CryptoView
+                      needsBackup={!isWalletBackedUp}
+                      onUpdateAccountName={onUpdateAccountName}
+                      onDoneViewingPrivateKey={onDoneViewingPrivateKey}
+                      onViewPrivateKey={onViewPrivateKey}
+                      defaultWallet={defaultWallet}
+                      onOpenWalletSettings={onOpenWalletSettings}
+                      isMetaMaskInstalled={isMetaMaskInstalled}
+                      sessionRoute={sessionRoute}
+                    />
+                  </Route>
+                }
+
+                {isWalletLocked && <Redirect to={WalletRoutes.Unlock} />}
+                {!isWalletLocked && <Redirect to={WalletRoutes.Portfolio} />}
+
+              </Switch>
```
New logic for handling page access

josheleonard commented on components/brave_wallet_ui/page/reducers/page_reducer.ts:
```diff
@@ -67,13 +70,15 @@ export const createPageReducer = (initialState: PageState) => {
-    const newState: PageState = {
+  reducer.on(Actions.walletSetupComplete, (state: PageState, payload?: boolean): PageState => {
+    // complete setup unless explicitly halted
+    const setupStillInProgress = !payload
+
+    return {
       ...state,
-      setupStillInProgress: false
+      mnemonic: undefined,
+      setupStillInProgress
```
We can now prevent onboarding completion after creating a wallet (when desired)

josheleonard replied:
The page "container" component has logic to restrict routes based off of if the wallet has been created yet. The old behavior would mark the setup process as complete when a wallet was created. This variable change would then change the accessible routes, blocking onboarding screen access, and redirecting to the portfolio screen. Now, I can optionally not mark the setup process as complete, so that the user can continue viewing onboarding screens for steps such as the verify seed phrase screen

kdenhartog replied:
Ok, that's interesting and I want to look into this a bit more then just to make sure this couldn't be used to bypass the wallet by re-onboarding and overriding the password or something similar to that

josheleonard commented on components/brave_wallet_ui/page/reducers/page_reducer.ts:
```diff
@@ -139,10 +144,15 @@ export const createPageReducer = (initialState: PageState) => {
 
-  reducer.on(Actions.setImportWalletError, (state: PageState, payload: ImportWalletError) => {
+  reducer.on(Actions.setImportWalletError, (state: PageState, {
+    hasError,
+    errorMessage,
+    incrementAttempts
+  }: ImportWalletErrorPayloadType) => {
     return {
       ...state,
-      importWalletError: payload
+      importWalletError: { hasError, errorMessage },
+      importWalletAttempts: incrementAttempts ? state.importWalletAttempts + 1 : state.importWalletAttempts
```
We can now track the import attempts so that we know when the operation completed

josheleonard commented on components/brave_wallet_ui/page/reducers/page_reducer.ts:
```diff
@@ -166,6 +176,21 @@ export const createPageReducer = (initialState: PageState) => {
+  reducer.on(Actions.agreeToWalletTerms, (state: PageState): PageState => {
+    return {
+      ...state,
+      walletTermsAcknowledged: true
```
Temporarily tracking that wallet terms were agreed to during onboarding

josheleonard commented on components/brave_wallet_ui/page/screens/fund-wallet/deposit-funds.tsx:
```diff
@@ -0,0 +1,453 @@
+import CreateAccountTab from '../../../components/buy-send-swap/create-account/index'
+import SelectHeader from '../../../components/buy-send-swap/select-header/index'
+
+export const DepositFundsScreen = () => {
```
Deposit crypto screen

josheleonard commented on components/brave_wallet_ui/page/screens/fund-wallet/fund-wallet.tsx:
```diff
@@ -0,0 +1,365 @@
+import SwapInputComponent from '../../../components/buy-send-swap/swap-input-component'
+import SelectHeader from '../../../components/buy-send-swap/select-header'
+
+export const FundWalletScreen = () => {
```
Screen for selecting crypto assets to buy via on-ramp. (opens link to on-ramp partners)

josheleonard commented on components/brave_wallet_ui/page/screens/onboarding/components/recovery-phrase/recovery-phrase.tsx:
```diff
@@ -0,0 +1,184 @@
+  return [...prevWords, word]
+}
+
+export const RecoveryPhrase: React.FC<Props> = ({
```
Component that can reveal the recovery phrase

josheleonard commented on components/brave_wallet_ui/page/screens/onboarding/create-password/onboarding-create-password.tsx:
```diff
@@ -0,0 +1,97 @@
+
+  // state
+  const [isValid, setIsValid] = React.useState(false)
+  const [password, setPassword] = React.useState('')
+
+  // methods
+  const nextStep = React.useCallback(() => {
+    if (isValid) {
+      dispatch(WalletPageActions.createWallet({ password }))
+      history.push(WalletRoutes.OnboardingExplainRecoveryPhrase)
+    }
+  }, [password, isValid])
+
+  const handlePasswordChange = React.useCallback(({ isValid, password }: NewPasswordValues) => {
+    setPassword(password)
+    setIsValid(isValid)
+  }, [])
+
+  // render
+  return (
+    <CenteredPageLayout>
+      <MainWrapper>
+        <StyledWrapper>
+
+          <OnboardingNewWalletStepsNavigation
+            goBackUrl={WalletRoutes.OnboardingWelcome}
+            currentStep={WalletRoutes.OnboardingCreatePassword}
+            preventSkipAhead
+          />
+
+          <TitleAndDescriptionContainer>
+            <Title>{getLocale('braveWalletCreatePasswordTitle')}</Title>
+            <Description>{getLocale('braveWalletCreatePasswordDescription')}</Description>
+          </TitleAndDescriptionContainer>
+
+          <NewPasswordInput
+            autoFocus={true}
+            onSubmit={nextStep}
+            onChange={handlePasswordChange}
+          />
```
Need to create a valid password to continue

josheleonard commented on components/brave_wallet_ui/page/screens/onboarding/disclosures/disclosures.tsx:
```diff
@@ -0,0 +1,152 @@
+  return <p key={text.duringTag}>
+    {text.beforeTag}
+    <LinkText
+      href='https://brave.com/terms-of-use/'
```
Terms to accept

josheleonard commented on components/brave_wallet_ui/page/screens/onboarding/disclosures/disclosures.tsx:
```diff
@@ -0,0 +1,152 @@
+  )
+
+  // memos
+  const isNextStepEnabled = React.useMemo(() => {
+    return isResponsibilityCheckboxChecked && isTermsCheckboxChecked
+  }, [isResponsibilityCheckboxChecked, isTermsCheckboxChecked])
```
Requirements to continue

josheleonard commented on components/brave_wallet_ui/panel/container.tsx:
```diff
@@ -226,16 +217,9 @@ function Container () {
-    if (hasIncorrectPassword) {
-      dispatch(WalletActions.hasIncorrectPassword(false))
-    }
+  const unlockWallet = (password: string) => {
+    dispatch(WalletActions.unlockWallet({ password }))
```
Input clearing no longer needed since the lock-screen component holds its own state now

josheleonard commented on components/brave_wallet_ui/utils/address-utils.ts:
```diff
@@ -32,3 +36,11 @@ export function isValidAddress (value: string, length: number): boolean {
   return accounts.some(account => account.deviceId && account.address === address)
 }
+
+export const suggestNewAccountName = (
+  accounts: WalletAccountType[],
+  network: BraveWallet.NetworkInfo
+) => {
+  const accountTypeLength = accounts.filter((account) => account.coin === network.coin).length + 1
+  return `${network.symbolName} ${getLocale('braveWalletAccount')} ${accountTypeLength}`
+}
```
Helper for creating a new account name

josheleonard commented on components/brave_wallet_ui/utils/asset-utils.ts:
```diff
@@ -14,20 +14,47 @@ export const getUniqueAssets = (assets: BraveWallet.BlockchainToken[]) => {
   }) !== -1
 }
 
+export const getWyreAssetSymbol = (asset: BraveWallet.BlockchainToken) => {
+  if (
+    !asset.contractAddress || // gas coins ok
+    asset.chainId === BraveWallet.MAINNET_CHAIN_ID // ETH-ERC coins ok
+  ) {
+    return asset.symbol
+  }
+
+  // format non-ethereum EVM token symbols for Wyre
+  const prefix = getWyreNetworkPrefix(asset.chainId)
+  return prefix ? `${prefix}${asset.symbol.toUpperCase()}` : asset.symbol
+}
+
 export const getRampAssetSymbol = (asset: BraveWallet.BlockchainToken) => {
   if (asset.symbol.toUpperCase() === 'BAT' && asset.chainId === BraveWallet.MAINNET_CHAIN_ID) {
     // BAT is the only token on Ethereum Mainnet with a prefix on Ramp.Network
     return 'ETH_BAT'
   }
 
+  if (asset.chainId === BraveWallet.AVALANCHE_MAINNET_CHAIN_ID && asset.contractAddress === '') {
+    return asset.symbol // AVAX native token has no prefix
+  }
```
Allow multichain purchases

josheleonard commented on components/brave_wallet_ui/utils/asset-utils.ts:
```diff
@@ -14,20 +14,47 @@ export const getUniqueAssets = (assets: BraveWallet.BlockchainToken[]) => {
   return rampNetworkPrefix !== '' ? `${rampNetworkPrefix}_${asset.symbol.toUpperCase()}` : asset.symbol
 }
+
+export const addLogoToToken = (token: BraveWallet.BlockchainToken) => {
+  return {
+    ...token,
+    logo: `chrome://erc-token-images/${token.logo}`
+  }
+}
```
use internal logos

josheleonard replied:
@kdenhartog check if chrome-untrusted is necessary

kdenhartog replied:
@josheleonard yeah let's use chrome-untrusted for this. It looks like we've migrated other PRs over to this approach as well for image rendering in the wallet

josheleonard replied:
Will we need to us the untrusted iframe for ALL image file extensions?

Also, can I use a query string in the iframe that contains the remote image URL, so that I do not need to send `postMessage`s to each iframe?

josheleonard replied:
@kdenhartog 

Standalone PR for this here: https://github.com/brave/brave-core/pull/14535
Standalone Security review: https://github.com/brave/security/issues/970

kdenhartog replied:
Thanks for breaking that one out

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/66af5d36a504edca7a9be1a8f2b9397d78f44074/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/d5a85237cdb7365d35274212e2fdbfda05b647ee/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/ba9378e17fb23464ade54cd196e61b2b619af076/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/b9d27b15f0aa51dc6c76b20fe964def7bfd88b5c/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/b9d27b15f0aa51dc6c76b20fe964def7bfd88b5c/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/49e6320243afb3c48a0e951515131f69bc791efe/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/0aa6b7cd38c35ac520bb517936e55c6f77fccf6a/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/0b23b8a304b41b4723bba7d14d931321978abf2a/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/c9c3010a1074b9e7cd4e72c025e6cb8d40ed4786/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/6892d4a20d9a981e2a69452c7dc90c2573d8eb69/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/2dd2ab19ff6c22ae21ba51cbc544cbc5facdbaf8/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/a1db6f9bf31bc781b580f16f423b9b9ea68d9a6c/storybook/index.html)

kdenhartog commented on components/brave_wallet_ui/page/screens/onboarding/components/recovery-phrase/recovery-phrase.tsx:
```diff
@@ -0,0 +1,184 @@
+
+const FAKE_PHRASE_WORDS = new Array(12).fill('Fake')
+
+const SELECTED_WORD_ORDINALS = {
```
Let's randomize which 3 words are chosen so we don't encourage a bad pattern of just remembering the first, third, last words rather than properly backing it up

josheleonard replied:
https://github.com/brave/brave-browser/issues/24683

kdenhartog commented on components/brave_wallet_ui/page/screens/onboarding/components/recovery-phrase/recovery-phrase.tsx:
```diff
@@ -0,0 +1,184 @@
+    const array = recoveryPhrase.slice().sort()
+
+    for (let i = array.length - 1; i > 0; i--) {
+      let j = unbiasedRandom(0, array.length - 1)
```
TODO @kdenhartog: Figure out what's happening with unbiased random and why it's being used vs a standard random function

kdenhartog commented on components/brave_wallet_ui/utils/asset-utils.ts:
```diff
@@ -14,20 +18,51 @@ export const getUniqueAssets = (assets: BraveWallet.BlockchainToken[]) => {
+export const addLogoToToken = (token: BraveWallet.BlockchainToken) => {
+  return {
+    ...token,
+    logo: token.logo?.startsWith('ipfs://')
```
Can we consolidate this logic into a helper function. I've seen it reused a few times across #13370 and #12510

josheleonard replied:
This logic was moved into the utils folder for this purpose. This is now exported for usage as a helper function

kdenhartog replied:
Sorry I meant the logo specific logic

kdenhartog replied:
Resolving this one for now, this is a small refactor just to prevent code reuse more than anything so not something that would block this

kdenhartog reviewed (commented):
Here's some initial static analysis of the code. Given the size of this PR and this being mainly UI focused changes I'm going to spend a fair amount of time focused on dynamic testing of this going forward as well

josheleonard commented on components/brave_wallet_ui/common/hooks/use-password-strength.ts:
```diff
@@ -4,23 +4,37 @@
-    setIsStrongPassword(isStrong)
+  // methods
+  const checkIsStrongPassword = React.useCallback((pass: string) => {
+    // is at least 8 characters
```
updated:  755f8a1ab57b8c6971e30c332492f83e0c1c0d56

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/755f8a1ab57b8c6971e30c332492f83e0c1c0d56/storybook/index.html)

kdenhartog commented on components/brave_wallet_ui/components/shared/password-input/password-strength-bar.tsx:
```diff
@@ -0,0 +1,73 @@
+  passwordStrength: PasswordStrengthResults
+}
+
+export const PasswordStrengthBar: React.FC<Props> = ({
```
![password indicator needs flexibility](https://user-images.githubusercontent.com/23125059/182286406-bc273586-64a2-48a8-a340-08b11d140891.gif)

Related to this it looks like the password now only either is strong or weak and there's no in between. Can we change this so it's relative to the length where 8 weak, 12 is medium, 16 is strong?

josheleonard replied:
> ![password indicator needs flexibility](https://user-images.githubusercontent.com/23125059/182286406-bc273586-64a2-48a8-a340-08b11d140891.gif) [ ![password indicator needs flexibility](https://user-images.githubusercontent.com/23125059/182286406-bc273586-64a2-48a8-a340-08b11d140891.gif) ](https://user-images.githubusercontent.com/23125059/182286406-bc273586-64a2-48a8-a340-08b11d140891.gif) [ ](https://user-images.githubusercontent.com/23125059/182286406-bc273586-64a2-48a8-a340-08b11d140891.gif)
> 
> Related to this it looks like the password now only either is strong or weak and there's no in between. Can we change this so it's relative to the length where 8 weak, 12 is medium, 16 is strong?

implemented here: 0cb8530a0bdedcf10623e7c883be3a9e24a071c9

kdenhartog commented:
https://user-images.githubusercontent.com/23125059/182286639-7af4a7fa-4e0e-4a5c-a298-0b864fb312db.mov

Small bug spotted. If the user clicks the link to the Terms of Use it automatically agrees to them.

kdenhartog commented:
nit: The spacing on the text here looks slightly lower than the designs in the spec

What's implemented
![Screen Shot 2022-08-02 at 3 53 44 PM](https://user-images.githubusercontent.com/23125059/182288274-8b7f6d25-6c60-44c6-8932-48e9db48dc57.png)

---

From the Spec:
![Screen Shot 2022-08-02 at 4 03 02 PM](https://user-images.githubusercontent.com/23125059/182289246-5a91c9c7-d21e-40b0-99f7-98a42a363186.png)

kdenhartog commented:
https://user-images.githubusercontent.com/23125059/182286639-7af4a7fa-4e0e-4a5c-a298-0b864fb312db.mov

Small bug spotted. If the user clicks the link to the Terms of Use it automatically agrees to them.

kdenhartog commented:
Found a bug where it seems that restore flow can't be used after the seed has been automatically generated. Instead it skips straight to setting the seed that was generated rather than letting the user provide the seed like normally done
![Weird mnemonic state skipping](https://user-images.githubusercontent.com/23125059/182294322-80015176-510c-4ceb-b9e6-e64cb6ac9de4.gif)
.

kdenhartog commented:
This PR reintroduces the [session state bug](https://twitter.com/HalbornSecurity/status/1537105573916860416) where the seed gets stored in the session file. The mitigation that was used to fix this last time was to use `autocomplete="off"`. See this commit for an example.

![Session file stores seed in plaintext](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif)

josheleonard commented:
> Screen.Recording.2022-08-02.at.3.16.08.PM.mov 
> Small bug spotted. If the user clicks the link to the Terms of Use it automatically agrees to them.

Fixed here: 3a397da79fbe19b24bedfaf9768bf4253bf18280

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/7ec13bbdd90835fede5e4c9edc1e7cb41f803e43/storybook/index.html)

josheleonard commented:
> This PR reintroduces the [session state bug](https://twitter.com/HalbornSecurity/status/1537105573916860416) where the seed gets stored in the session file. The mitigation that was used to fix this last time was to use `autocomplete="off"`. See this [commit](https://github.com/brave/brave-core/commit/33086b43a2a444099a18b201e293c60025cf5e1d) for an example.
> 
> ![Session file stores seed in plaintext](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif) [ ![Session file stores seed in plaintext](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif) ](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif) [ ](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif)

Good catch! Fixed here: 7ec13bbdd90835fede5e4c9edc1e7cb41f803e43

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/f7f8a52bef622121d7d1471deefdbf46662bd652/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/bd179b57247a00b5a6b8a4527f1b33e03fc6402e/storybook/index.html)

josheleonard commented on components/brave_wallet_ui/constants/types.ts:
```diff
@@ -612,6 +634,7 @@ export enum WalletRoutes {
   // portfolio
   Portfolio = '/crypto/portfolio',
   PortfolioAsset = '/crypto/portfolio/:id/:tokenId?',
+  PortfolioSub = '/crypto/portfolio/:id?',
```
a sub-page. nested route

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/0cb8530a0bdedcf10623e7c883be3a9e24a071c9/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/a680eeb08d5fcd1b3696cbf963db2b76e206e07e/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/414f65b6c06da01276a2c55e55332ef42213e2b2/storybook/index.html)

josheleonard commented on components/brave_wallet_ui/components/shared/create-placeholder-icon/index.tsx:
```diff
@@ -75,7 +75,7 @@ function withPlaceholderIcon (WrappedComponent: React.ComponentType<any>, config
     const remoteImage = React.useMemo(() => {
       if (isRemoteURL) {
-        return httpifyIpfsUrl(tokenImageURL)
+        return `chrome://image?${httpifyIpfsUrl(tokenImageURL)}`
```
PR created for this here: https://github.com/brave/brave-core/pull/14535

josheleonard commented:
> This PR reintroduces the [session state bug](https://twitter.com/HalbornSecurity/status/1537105573916860416) where the seed gets stored in the session file. The mitigation that was used to fix this last time was to use `autocomplete="off"`. See this [commit](https://github.com/brave/brave-core/commit/33086b43a2a444099a18b201e293c60025cf5e1d) for an example.
> 
> ![Session file stores seed in plaintext](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif) [ ![Session file stores seed in plaintext](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif) ](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif) [ ](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif)

Seems like the textarea was also causing this regression. Should be fixed here: 6499242907bd4afd79e45d098d3b5caebb4c8ea5

cc @kdenhartog

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/6499242907bd4afd79e45d098d3b5caebb4c8ea5/storybook/index.html)

Douglashdaniel reviewed (commented):
Do we need the `Asset` filter selector here?

https://user-images.githubusercontent.com/40611140/184213045-49b61e76-42e8-4dd9-a70c-a0b68d8a3afc.mov

Douglashdaniel reviewed (commented):
Not sure what this warning is about

https://user-images.githubusercontent.com/40611140/184215752-ff3d0299-b5af-441d-9b94-6f5221b9596f.mov

Douglashdaniel reviewed (commented):
Clicking `Deposit funds` is locking up the page.

https://user-images.githubusercontent.com/40611140/184217511-663eb5dd-7551-4456-af15-e03ae0349743.mov

kdenhartog reviewed (approved):
@thypon has taken a look at this dynamically and I have done a static analysis review and limited dynamic testing. We both think this should be fine from a security perspective so approving and will close the issue. I do not believe there's any new privacy considerations that are needed here either.

josheleonard commented:
> Do we need the `Asset` filter selector here?
> 
>  Screen.Recording.2022-08-11.at.1.26.22.PM.mov

Removed. Fixed here: b3defed10a7d530b42a3e5c64d90d0d02524c944

kdenhartog commented:
> > This PR reintroduces the [session state bug](https://twitter.com/HalbornSecurity/status/1537105573916860416) where the seed gets stored in the session file. The mitigation that was used to fix this last time was to use `autocomplete="off"`. See this [commit](https://github.com/brave/brave-core/commit/33086b43a2a444099a18b201e293c60025cf5e1d) for an example.
> 
> > 
> 
> > ![Session file stores seed in plaintext](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif) [ ![Session file stores seed in plaintext](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif) ](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif) [ ](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif)
> 
> 
> 
> Seems like the textarea was also causing this regression. Should be fixed here: 6499242907bd4afd79e45d098d3b5caebb4c8ea5
> 
> 
> 
> cc @kdenhartog

@josheleonard just thought of this now, but can we double check we're doing the same setup of the seed phrase on the password checks as well here? I'm away at a conference for this week so won't be able to check until next week but should be the same thing as we did for the seed

(and the new ones we've been adding to the backup page)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/b3defed10a7d530b42a3e5c64d90d0d02524c944/storybook/index.html)

josheleonard commented:
> 

I verified that the password fields are not leaking info to the session. The inputs for password are input fields with `autoComplete='off'`

Douglashdaniel reviewed (approved):
++ Front-end

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/4e9faed851b54d548a909e16aa705c94ad5e9802/storybook/index.html)

brave-builds commented:
A Storybook has been deployed to [preview UI for the latest push](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/feat-onboarding-next/10651b439e83605ba08b49a34d29259b5d746ec8/storybook/index.html)

spylogsster commented:
> This PR reintroduces the [session state bug](https://twitter.com/HalbornSecurity/status/1537105573916860416) where the seed gets stored in the session file. The mitigation that was used to fix this last time was to use `autocomplete="off"`. See this [commit](https://github.com/brave/brave-core/commit/33086b43a2a444099a18b201e293c60025cf5e1d) for an example.
> 
> ![Session file stores seed in plaintext](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif) [ ![Session file stores seed in plaintext](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif) ](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif) [ ](https://user-images.githubusercontent.com/23125059/182295897-a9ec7e85-264a-44dd-b7bc-47944c0c0373.gif)

https://bravesoftware.slack.com/archives/CHGKGMHDJ/p1661367744020979 Fix for sessions has been uplifted, cc @bsclifton @kdenhartog
